### PR TITLE
RavenDB-9721 and RavenDB-9723

### PR DESCRIPTION
**Pull request by arekpalinski**

### CODE REVIEW

File: src/Raven.Client/Documents/Session/Operations/QueryOperation.cs [hunk 1]
```diff
@@ -165,34 +165,39 @@ public void EnsureIsAcceptableAndSaveResult(QueryResult result)
...
+
+            _currentQueryResults = result;
+            //_currentQueryResults.EnsureSnapshot();
+
+            if (Logger.IsInfoEnabled)
             {
-                _sp?.Stop();
-                var msg = $"Waited for {_sp?.ElapsedMilliseconds:#,#;;0} ms for the query to return non stale result.";
+                var isStale = result.IsStale ? "stale " : "";
+                Logger.Info($"Query returned {result.Results.Items.Count()}/{result.TotalResults} {isStale}results");
```
Comment (ayende) [inline src/Raven.Client/Documents/Session/Operations/QueryOperation.cs:176]: Better log message here.

File: src/Raven.Client/Documents/Session/Operations/QueryOperation.cs [hunk 2]
```diff
@@ -165,34 +165,39 @@ public void EnsureIsAcceptableAndSaveResult(QueryResult result)
             if (result == null)
                 throw new IndexDoesNotExistException("Could not find index " + _indexName);
 
-            if (_indexQuery.WaitForNonStaleResults && result.IsStale)
+            EnsureIsAcceptable(result, _indexQuery.WaitForNonStaleResults, _sp, _session);
+
+            _currentQueryResults = result;
+            //_currentQueryResults.EnsureSnapshot();
```
Comment (ayende) [inline src/Raven.Client/Documents/Session/Operations/QueryOperation.cs:171]: why is this commented?

Comment (arekpalinski) [reply src/Raven.Client/Documents/Session/Operations/QueryOperation.cs:171]: no longer valid code - removed in https://github.com/ravendb/ravendb/pull/4681

File: src/Raven.Client/Documents/Session/Operations/QueryOperation.cs (hunk 1 above)
Comment (arekpalinski) [reply src/Raven.Client/Documents/Session/Operations/QueryOperation.cs:176]: fixed in https://github.com/ravendb/ravendb/pull/4681

**Pull request merged**